### PR TITLE
pkgbuild revision

### PR DESCRIPTION
**Pull request by FabioLolix**
Among others things using sudo and installing in users system instead of the package is prohibited

### CODE REVIEW

**Review by shibumi — APPROVED** [review]

**Comment by shibumi** [comment]
@FabioLolix  this is fine, but it 'changes' the package from being a source package to a precompiled binary package. 
I would suggest we add another folder with your PKGBUILD for the binary package.

What about the following structure:

```
aur-falco:
  README.md
  falco-bin/PKGBUILD
  falco/PKGBUILD 
```

EDIT: How do you guarantee that the precompiled binary works with the compiled Arch Linux kernel? Is it really full ABI compatible?

File: PKGBUILD [hunk 1]
```diff
@@ -1,62 +1,41 @@
...
-provides=()
-conflicts=()
-replaces=()
+provides=(falco)
+conflicts=(falco)
 backup=()
-options=()
-install=
-changelog=
+source_x86_64=("https://download.falco.org/packages/bin/${arch}/falco-${pkgver}-x86_64.tar.gz")
```
Comment (shibumi) [inline PKGBUILD:15]: How do you guarantee that the precompiled binary by falco works with the Arch Linux kernel?

**Review by shibumi — CHANGES_REQUESTED** [review]

**Comment by FabioLolix** [comment]
The package actually is already a binary package, I haven't actually changed the source=()

I'm working on a from source pkgbuild too

> What about the following structure:
> 
> ```
> aur-falco:
>   README.md
>   falco-bin/PKGBUILD
>   falco/PKGBUILD 
> ```

don't know if you want to make 2 separate repository or keep one for each that can push to AUR,

**Comment by shibumi** [comment]
@FabioLolix  why do you think it's a binary package? Kris is explicitly calling `make` here:
```
build() {
    echo "Compiling Kernel Module..."
    cd $pkgname-$pkgver-$arch/usr/src/falco*
    make
```

??

**Comment by FabioLolix** [comment]
> @FabioLolix why do you think it's a binary package? Kris is explicitly calling `make` here:
> 
> ```
> build() {
>     echo "Compiling Kernel Module..."
>     cd $pkgname-$pkgver-$arch/usr/src/falco*
>     make
> ```

That is for the dkms module

**Comment by shibumi** [comment]
> That is for the dkms module

a dkms module works completely different in Arch Linux. For a dkms module you have to place a hook into the package and a dkms.conf. See here: https://github.com/archlinux/svntogit-community/blob/f9a512e603615566440b170995439cf626f0836d/trunk/dkms.conf

**Review by shibumi — CHANGES_REQUESTED** [review]
not convinced yet. I suggest putting the -bin package into a separate directory

**Comment by FabioLolix** [comment]
> a dkms module works completely different in Arch Linux. For a dkms module you have to place a hook into the package and a dkms.conf.

I'm not familiar with dkms modules but you can see that in my revision the module is not compiled in the pkgbuild, dkms.conf is installed in "${pkgdir}/usr/src/falco-${pkgver}" alongside source and Makefile

Splitting into bin + dkms is a best practice I guess

**Comment by krisnova** [comment]
this LGTM - I will merge now and push an update to the official repository shortly

Once my rack is back online we will have automation in place as well 

Thanks for the contribution

**Comment by shibumi** [comment]
> Splitting into bin + dkms is a best practice I guess 

yep, that's how we do this for the official packages, too :)

**Pull request merged**